libavfilter/vf_drawtext.c

```diff
@@ -731,9 +745,24 @@ static av_cold int init(AVFilterContext *ctx)
             s->text = av_strdup("");
     }
 
+    if (s->text_source_string) {
+        s->text_source = text_source_string_parse(s->text_source_string);
+        if ((int)s->text_source < 0) {
+            av_log(ctx, AV_LOG_ERROR, "Error text source: %s\n", s->text_source_string);
+            return AVERROR(EINVAL);
+        }
+    }
+
+    if (s->text_source == AV_FRAME_DATA_DETECTION_BBOXES) {
+        s->text = av_mallocz(AV_DETECTION_BBOX_LABEL_NAME_MAX_SIZE *
+                             (AV_NUM_DETECTION_BBOX_CLASSIFY + 1));
+        if (!s->text)
```

Comment:
it has been checked in line 763

Reply:
yes, but i'm a little confused, if it failed to allocate the mem, the code will still report err like line 765 said. But actually, it should report something like failed to allocate mem right?

Reply:
yes, and the change in #765 is no longer needed.

Reply:
Well, I think it may be  necessary. Since if not mem allocated successfully in  line 757, it will return `AVERROR(ENOMEM)`. And if no text/file/timecode/text_source provided, it will return `AVERROR(EINVAL)` and report error with in error info in line 765, which implicy that text_source is a optional text data source. These two kind `if (!s->text)` check seem not redundant.
What do you think?

Reply:
yes, you are right.

Reply:
OK, thank you, modified as we talked.